core/lib/dal/src/eth_watcher_dal.rs

```diff
@@ -0,0 +1,152 @@
+use zksync_db_connection::{connection::Connection, error::DalResult, instrument::InstrumentExt};
+use zksync_types::SLChainId;
+
+use crate::Core;
+
+pub struct ProcessedEventsDal<'a, 'c> {
```

Comment:
nit: let's make file name and struct name same. IMHO EthWatcherDal is better than ProcessedEventsDal